pkg/sdkserver/sdkserver.go

```diff
@@ -335,7 +327,7 @@ func (s *SDKServer) updateState(ctx context.Context) error {
 
 	// If we are setting the Allocated status, set the last-allocated annotation as well.
 	if gsCopy.Status.State == agonesv1.GameServerStateAllocated {
-		ts, err := s.clock.Now().MarshalText()
+		ts, err := s.clock.Now().UTC().MarshalText()
```

Comment:
if we're going to do this here, we also need to do it in Allocate() functionality as well, so it's UTC everywhere.

Reply:
I think all of the failures can be attributed to `TestPlayerConnectWithCapacityZero`, which I have #3062 out for. The remaining failures are unreported status because that test exploded. I'm seeing failures in that test in other builds as well, so I'm not certain it's this PR, and I did try to repro with that test locally, but let me try again.

Reply:
Let me revert this part.